### PR TITLE
Implements storage of tokens

### PR DESCRIPTION
**Pull request by VishalZ123**
feat: Implement Storage class for handling token storage and retrieval
Implements #5 
This commit adds a new class called `Storage` that provides methods for storing, retrieving and deleting a token using `AsyncStorage` in React Native. The `Storage` class is implemented in TypeScript and is designed to be reusable across the app.

The `storeData`, `getData`, and `removeData` methods of the `Storage` class allow the token to be saved to local storage, retrieved when needed, and deleted when no longer required (as demonstrated in the case of Logout).

In addition, the commit demonstrates how to use the `Storage` class to retrieve the token from local storage and set it to a state variable using the `useState` hook. This enables the token to be used in subsequent API calls to the backend.

Overall, this commit stores the token and makes it easier to manage the token across different parts of the app.

https://github.com/glific/mobile/assets/76409986/be3fd26d-9ef3-4ef5-bc74-07b93019de94

PS: The chat screen currently displays the token and a log out button. These can be removed when one implements the chat screen.

### CODE REVIEW

File: screens/Login.tsx [hunk 1]
```diff
@@ -44,7 +45,11 @@ const Login = ({ navigation }: Props) => {
         }
       })
 
-      console.log(response.data.data);
+      // console.log(response.data.data);
+      await Storage.storeData('token', response.data.data['access_token']);
+      await Storage.storeData('renewal_token', response.data.data['renewal_token']);
+      await Storage.storeData('token_expiry_time', response.data.data['token_expiry_time']);
+
```
Comment (mdshamoon) [inline screens/Login.tsx:52]: Let's store the whole object in a single JSON string

```suggestion
      await Storage.storeData('session', JSON.stringify(response.data.data);
```

File: screens/Chat.tsx [hunk 2]
```diff
@@ -1,9 +1,49 @@
...
+      setToken(tokenValue);
+    };
+
+    getToken();
+  }, []);
+
+  const LogoutHandler = async () => {
+    await Storage.removeData("token");
+    await Storage.removeData("renewal_token");
+    await Storage.removeData("token_expiry_time");
```
Comment (mdshamoon) [inline screens/Chat.tsx:29]: Update here based on the previous suggestion of storing in a single JSON string.

File: screens/Chat.tsx [hunk 3]
```diff
@@ -1,9 +1,49 @@
...
+    setToken("");
+    navigation.navigate("Login");
+    // const getToken = async () => {
+    //   const tokenValue = await Storage.getData("token");
+    //   setToken(tokenValue);
+    //   console.log("Token val-", tokenValue)
+    // };
+    // getToken();
```
Comment (mdshamoon) [inline screens/Chat.tsx:37]: Remove comments

File: tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "expo/tsconfig.base",
   "compilerOptions": {
-    "strict": true
+    "strict": true,
+    "allowUmdGlobalAccess": true,
```
Comment (mdshamoon) [inline tsconfig.json:5]: Do we really need this?

**Review by mdshamoon — CHANGES_REQUESTED** [review]
Added some comments, please check. Also please resolve the conflicts.

**Comment by VishalZ123** [comment]
Hi @mdshamoon 
I have made the requested changes as follows:
Now the session is stored as string and can be read.
Removed the comments
Reverted the tsconfig.json, file I changed it as there were some issues for me running it locally, I thought someone else might face the same issue.
Please review the updated code at your convenience. Let me know if there is anything else I can assist with.

**Comment by ajaman190** [comment]
It would be better if we used Redux Toolkit for single-session state management and Redux Persist along with Async Storage to store the user access token even after the session is closed.

File: tsconfig.json (hunk 4 above)
Comment (VishalZ123) [reply tsconfig.json]: It allow the use of `React` as a global variable. Might be some issue with my setup, so I added this. I have reverted it back.

**Comment by mdshamoon** [comment]
@ajaman190 for now its fine. We can always add if needed.

**Review by mdshamoon — APPROVED** [review]
Looks good. Please resolve the conflicts. Will merge after that.

**Pull request merged**